### PR TITLE
fixes for lagging scroll on process and fix IE11 performance

### PR DESCRIPTION
**Pull request by barteksiudak**

### CODE REVIEW

File: src/alto-ui/helpers/sum.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default function(arr, indexName) {
+  if (!Array.isArray(arr)) {
+    return 0;
+  }
+
+  return arr.reduce((sum, item) => {
+    const value = indexName ? item[indexName] : parseInt(item, 10);
+    
+    return sum + value || 0;
```
Comment (NicolasMartos) [inline src/alto-ui/helpers/sum.js]:
```suggestion
    const value = indexName ? item[indexName] : item;

    return sum + parseInt(value,10) || 0;
```

File: src/alto-ui/helpers/__tests__/sum.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  });
+
+  it('returns sum of strings and numbers array without second argument', () => {
+    expect(sum(['a', 'b', 1])).toBe(1);
+  });
+
+  it('returns sum of object array', () => {
+    const mock = [{ width: 1 }, { width: 2 }, { width: 3 }];
+    expect(sum(mock, 'width')).toBe(6);
+  });
```
Comment (NicolasMartos) [inline src/alto-ui/helpers/__tests__/sum.test.js]: Could you add a `it` without int ? 
`const mock = [{ width: '1' }, { width: '2' }, { width: '3' }];`

File: src/alto-ui/helpers/sum.js (hunk 1 above)
Comment (barteksiudak) [reply src/alto-ui/helpers/sum.js]: good catch

File: src/alto-ui/Datagrid/Datagrid.js [hunk 3]
```diff
@@ -91,56 +91,48 @@ class Datagrid extends React.PureComponent {
...
     this.handleToggleGroup = this.handleToggleGroup.bind(this);
     this.handleStopResize = this.handleStopResize.bind(this);
     this.handleStartResize = this.handleStartResize.bind(this);
-    this.trackDimensions = debounce(this.trackDimensions.bind(this), WINDOW_RESIZE_DEBOUNCE);
 
     this.containerRef = React.createRef();
+
+    this.setScrollNode = this.setScrollNode.bind(this);
+    this.setScrollHeaderNode = this.setScrollHeaderNode.bind(this);
+    this.scrollListener = this.isMacOS
```
Comment (kevinmlsilva) [inline src/alto-ui/Datagrid/Datagrid.js]: do we need to put the condition here? can't we have on on everything on this.scrollListener.bind(this) ?

File: src/alto-ui/Datagrid/Datagrid.js [hunk 4]
```diff
@@ -462,6 +426,32 @@ class Datagrid extends React.PureComponent {
     );
   }
 
+  renderHorizontalScroll(rowsWidth = 0) {
+    if (this.isMacOS) {
```
Comment (kevinmlsilva) [inline src/alto-ui/Datagrid/Datagrid.js]:
```suggestion
    if (this.isMacOS || !rowsWidth) {
```

File: src/alto-ui/Datagrid/Datagrid.js [hunk 5]
```diff
@@ -462,6 +426,32 @@ class Datagrid extends React.PureComponent {
...
+      return null;
+    }
+    const {
+      offsetWidth: frozenRowsWidth = 0,
+    } = this.frozenRowsNode || {};
+
+    if (!rowsWidth) {
```
Comment (kevinmlsilva) [inline src/alto-ui/Datagrid/Datagrid.js]: remove this condition

File: src/alto-ui/Datagrid/Datagrid.js (hunk 3 above)
Comment (barteksiudak) [reply src/alto-ui/Datagrid/Datagrid.js]: IMO it's a better place.
Constructor runs only once and this condition will be checked once.
In your proposition condition will be checked on scroll

File: src/alto-ui/helpers/__tests__/sum.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import sum from '../sum';
+
+describe('sum', () => {
```
Comment (NicolasMartos) [inline src/alto-ui/helpers/__tests__/sum.test.js]:
```suggestion
describe('/helpers/sum', () => {
```

File: src/alto-ui/helpers/__tests__/sum.test.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import sum from '../sum';
+
+describe('sum', () => {
+  it('is sum', () => {
```
Comment (NicolasMartos) [inline src/alto-ui/helpers/__tests__/sum.test.js]:
```suggestion
  it('should return 0 without params', () => {
```

File: src/alto-ui/helpers/__tests__/sum.test.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import sum from '../sum';
+
+describe('sum', () => {
+  it('is sum', () => {
+    expect(sum()).toBe(0);
+  });
+
+  it('returns sum of numbers array without second argument', () => {
```
Comment (NicolasMartos) [inline src/alto-ui/helpers/__tests__/sum.test.js]:
```suggestion
  it('should return sum values of an array if there is no second argument', () => {
```

File: src/alto-ui/Datagrid/Datagrid.js [hunk 9]
```diff
@@ -91,56 +91,48 @@ class Datagrid extends React.PureComponent {
...
       collapsedGroups: {},
       resizer: initialStateResizer,
       columnsWidth: {},
+
     };
 
-    this.scrollingElt = null;
-    this.timeout = null;
-    this.initialTimeout = null;
+    this.isMacOS = isMacOSHelper();
```
Comment (NicolasMartos) [inline src/alto-ui/Datagrid/Datagrid.js]: Could you create a const before definition of your component?

File: src/alto-ui/Datagrid/Datagrid.js [hunk 10]
```diff
@@ -462,6 +426,29 @@ class Datagrid extends React.PureComponent {
     );
   }
 
+  renderHorizontalScroll(rowsWidth = 0) {
```
Comment (NicolasMartos) [inline src/alto-ui/Datagrid/Datagrid.js]: so you create your own scrollbar except on macos?

Comment (barteksiudak) [reply src/alto-ui/Datagrid/Datagrid.js]: yes, on MacOS we don't see a scrollbar. We wan't only scroll by touch pad.

File: src/alto-ui/helpers/__tests__/sum.test.js (hunk 7 above)
Comment (barteksiudak) [reply src/alto-ui/helpers/__tests__/sum.test.js]: I'm testing by TDD so the first assertion is always default value.
When I wrote that test `sum` function doesn't exists yet.
I think that assertion tells us everything what we want to know in this case `expect(sum()).toBe(0);`

File: src/alto-ui/Datagrid/Datagrid.js [hunk 11]
```diff
@@ -501,51 +482,68 @@ class Datagrid extends React.PureComponent {
...
+            </div>
           </div>
         </div>
+        <div
+          id={id}
+          role="grid"
+          aria-rowcount={headersCount + rows.length}
+          className={bemClass(
+            'Datagrid',
+            { loaded: true, 'with-summary': !!this.props.renderSummaryCell },
```
Comment (NicolasMartos) [inline src/alto-ui/Datagrid/Datagrid.js]: That one seems to not be necessary anymore

Comment (barteksiudak) [reply src/alto-ui/Datagrid/Datagrid.js]: Good point, I wondered where is this hidden. I found the opacity: 1 :)
I will change it now

**Comment by NicolasMartos** [comment]
Could you explain into zendesk ticket impact we can have on ProcessGrid and Datagrid to help testers?

**Comment by barteksiudak** [comment]
> Could you explain into zendesk ticket impact we can have on ProcessGrid and Datagrid to help testers?

sure

**Pull request merged**